### PR TITLE
Add test for empty initializer. Document other tests.

### PR DESCRIPTION
**Pull request by rabbah**
Shifts tests which are part of the system package (but slated for removal as part of #3737, #3840) to container unit tests which also checks that the runtimes provide a consistent handling of different capabilities and error scenarios.

## Related issue and scope
- [x] I opened an issue to propose and discuss this change (#3839, #3737)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [x] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

File: tests/src/test/scala/actionContainers/BasicActionRunnerTests.scala [hunk 1]
```diff
@@ -19,14 +19,32 @@ package actionContainers
...
 
 @RunWith(classOf[JUnitRunner])
 trait BasicActionRunnerTests extends ActionProxyContainerTestUtils {
   def withActionContainer(env: Map[String, String] = Map.empty)(code: ActionContainer => Unit): (String, String)
 
+  /**
+   * Runs tests for actions which receive an empty initializer (no source or exec).
+   * @param checkResultInLogs should be true iff the result of the action is expected to appear in stdout or stderr
+   */
+  def testNoSourceOrExec(checkResultInLogs: Boolean = true) = {
```
Comment (dgrove-oss) [inline tests/src/test/scala/actionContainers/BasicActionRunnerTests.scala]: Seems like checkResultInLogs is not used....should it be removed?

Comment (rabbah) [reply tests/src/test/scala/actionContainers/BasicActionRunnerTests.scala]: updated to remove the relevant code.
